include/CMakeLists.txt

```diff
@@ -10,3 +10,18 @@ target_sources(VulkanLayerSettings PRIVATE
 	vulkan/layer/vk_layer_settings.hpp
 	vulkan/layer/vk_layer_settings_ext.h
 )
+
+
+add_library(VulkanLayerDispatchTable INTERFACE)
+add_library(Vulkan::LayerDispatchTable ALIAS VulkanLayerDispatchTable)
+
+target_sources(VulkanLayerDispatchTable PRIVATE
+    vulkan/layer/vk_layer_dispatch_table.h
+    vulkan/layer/vk_dispatch_table_helper.h
+)
+
+# NOTE: Because Vulkan::Headers header files are exposed in the public facing interface
+# we must expose this library as public to users.
+target_link_Libraries(VulkanLayerDispatchTable INTERFACE Vulkan::Headers)
+
+target_include_directories(VulkanLayerDispatchTable INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
```

Comment:
This made me realize `vk_enum_string_helper.h` doesn't really have a dedicated library right now...

Reply:
Should there be a 'everything' target? Like, imagine we have 5 utilities in the repo, would it make sense to create a new target which has those 5 utilities as public dependencies? This is putting the cart before the horse, so best to not do anything like that until we have people complaining that there are too many targets to link.

Reply:
I think a header target would be appropriate.

`Vulkan::UtilityHeaders` that way it's also clear this target is only comprised of headers files.

Reply:
I may do that as there are a few extra bits that Vulkan-Profiles needs to be able to use the LayerDispatchTable (cause there is *other* stuff that profiles uses which aren't the table).